File: 049-Rearrangement of Bits/RearrangementofBits.py
def rearrange_bits(n):
    # Count set bits in binary representation
    count = bin(n).count('1')
    # Create minimum number with that many set bits (e.g., 0011)
    return (1 << count) - 1

# Test cases
print(rearrange_bits(10))  # Output: 3
print(rearrange_bits(2))   # Output: 1

File: 049-Rearrangement of Bits/RearrangementofBits.cpp
#include<iostream>
using namespace std;

int rearrangeBits(int n){
    int count = 0;

    while (n > 0) {
        if (n & 1) { // Check if the last bit is set
            count++;
        }
        n >>= 1; // Right shift to process the next bit
    }

    int result = (1 << count) - 1; // Create a number with 'count' bits set to 1
    return result;
}

int main() {
    int n = 10;
    cout << rearrangeBits(n) << endl; // Output: 3

    n = 2;
    cout << rearrangeBits(n) << endl; // Output: 1

    return 0;
}